VA-Explorer/va_explorer
# Fieldworker usernames

This pull request partially implements [Pivotal Story, Ability to tie VAs to interviewers](https://www.pivotaltracker.com/story/show/177311007). I am breaking this ticket up into multiple pull requests. 

**This pull request implements the first part of this story:**
"Each VA Explorer interviewer user account should be able to be tied to a username (we may eventually want to support more than one username, but start with one) on the user management page"

**This major components of this pull request are:**
- Adds an optional "username" field (va_username in the database) to the user creation and update forms that is only shown when the Field Worker role is selected
- Adds a "username" field to the UserDetail page, that is shown only for Field Worker users
- Adds the model VaUsername that has a ForeignKey to the user and a va_username field (this field will be joined on the username field in va_data_management_verbalautopsy ) **Note:** I am not calling this field "username" in the model to avoid confusion with the username field that exists on the user model and which comes automatically from django.contrib.auth. 
- Implements the va_username field as one-to-one with the User through logic in users/forms.py. This logic can be updated to accommodate having more than one va_username per Field Worker if needed.
- Updates tests to test for correctness

**To test this functionality in the UI:**
1. As an admin user, confirm that the Username field is only visible on the User Create and Update forms when the Field Worker role is selected
2. As an admin user, confirm that the Username field is optional when creating a Field Worker
3. As an admin user, confirm that the Username field is optional when updating a Field Worker
4. As an admin user, confirm that you can assign a Username to a Field Worker and that the Username appears on their UserDetail page
5. As an admin user, confirm that you cannot assign the same Username to two different Field Workers 
6. As an admin user, confirm that updating a Field Worker with a preexisting Username and leaving the Username field blank on update, removes the Username
